draft-ietf-ice-pac.xml

```diff
@@ -180,7 +180,17 @@
         equal to the agent's connectivity check transaction timeout.
       </t>
       <t>
-        This timeout value is chosen to roughly coincide with the maximum
+        Specifically, the ICE agent will start its timer once it is ready to
+        send and receive ICE connectivity checks. This occurs when the agent has
+        sent and received the values needed to perform connectivity checks
+        (e.g., the Username Fragment and Password denoted in
+        <xref target="RFC8445" />, Section 5.3), and has sent at least one ICE
+        candidate. This ensures that the timer is only started once the remote
+        peer is actively trying to connect (i.e., to the ICE candidates that
+        were sent).
```

Comment:
Based on the text, the answerer will start the timer once it has received the offer and sent an answer with at least one ICE candidate. So far so good.

However, being strict, at that point the remote peer is yet not "trying actively to connect", because it needs to wait for the answer before it can do that (assuming you by "trying to connect" mean sending of connectivity checks).

Reply:
The remote peer sent the offer, ergo it is actively trying to connect. The answer could be delayed for many reasons, which is why the offerer needs to wait.

Reply:
That is not aligned with the other text, that says:

"Specifically, the ICE agent will start its timer once it is ready to
 send and receive ICE connectivity checks. This occurs when the agent has
 sent and received the values needed to perform connectivity checks"

That text does not only talk about having sent the offer, but also having received the answer (which will contain the values needed to perform connectivity checks).

Do we really need the "ensure" sentence?

Reply:
I don't think there's any contradiction; the offerer doesn't start its timer until it gets the answer.  I also think it's important to explain why this particular instant was chosen.

Reply:
At least the text needs to be modified. ICE is all about establishing data plane connectivity, so when I read "actively trying to connect" I think about sending connectivity checks.

Reply:
Understood. Let me see if I can tweak the text accordingly.

Reply:
PTAL

Reply:
Looks better. Just one thing:

Assume the remote peer is full trickle. It may not provide any candidates, but as far as I know it will still provide e.g., the username and password.

However, in the text it is not clear whether "the values needed to perform connectivity checks" includes candidates or not. Because, when looking at the referenced section 5.3 of 8445, candidates ARE listed as "values". 

Perhaps we would need to clarify that an ICE agent will start the timer even if it has not yet received a candidate from the remote peer, as long as it has received other "values values needed to perform connectivity checks".

Reply:
Good point, will clarify.

Reply:
I made the requested changes, but while thinking about this I realized that we haven't specified what to do if no candidates are sent. There are some edge cases where we would still want the timer even if we don't send any candidates (e.g., we receive a single candidate from the peer but still want to discover prflx candidates from the peer), so I relaxed the text to not require sending a candidate.

Reply:
I commented on this below. With the current text, we can start the timer when NO endpoint has yet provided any comments. I assume that can happen if both endpoints are full trickle?